blocks/bricks/sequence_generators.py

```diff
@@ -94,36 +94,43 @@ class BaseSequenceGenerator(Initializable):
 
     """
     @lazy
-    def __init__(self, readout, transition, fork=None, **kwargs):
+    def __init__(self, readout, transition, fork, **kwargs):
         super(BaseSequenceGenerator, self).__init__(**kwargs)
         self.readout = readout
         self.transition = transition
         self.fork = fork
 
-        self.state_names = transition.compute_states.outputs
-        self.context_names = transition.apply.contexts
-        self.glimpse_names = transition.take_glimpses.outputs
         self.children = [self.readout, self.fork, self.transition]
 
+    @property
```

Comment:
I you want these to be properties, better make them private in order to force the user use `generate.states`, `generate.contexts`, etc.. There should not be two alternative interfaces to query such things, and this was merely a shortcut.